crates/pera-core/src/dwallet_mpc/mpc_session.rs

```diff
@@ -143,12 +132,132 @@ impl DWalletMPCSession {
         }
     }
 
+    fn advance_specific_party(
+        &self,
+    ) -> DwalletMPCResult<AsynchronousRoundResult<Vec<u8>, Vec<u8>, Vec<u8>>> {
+        let session_id = CommitmentSizedNumber::from_le_slice(
+            self.session_info.flow_session_id.to_vec().as_slice(),
+        );
+        match &self.session_info.mpc_round {
+            MPCRound::DKGFirst => {
+                let public_input = bcs::from_bytes(&self.public_input)?;
+                crate::dwallet_mpc::advance::<DKGFirstParty>(
+                    session_id,
+                    self.party_id,
+                    &self.weighted_threshold_access_structure,
+                    self.pending_messages.clone(),
+                    public_input,
+                    (),
+                )
+            }
+            MPCRound::DKGSecond(event_data, _) => {
+                let public_input = bcs::from_bytes(&self.public_input)?;
+                let result = crate::dwallet_mpc::advance::<DKGSecondParty>(
+                    session_id,
+                    self.party_id,
+                    &self.weighted_threshold_access_structure,
+                    self.pending_messages.clone(),
+                    public_input,
+                    (),
+                )?;
+                if let AsynchronousRoundResult::Finalize { public_output, .. } = &result {
```

Comment:
Please... add documentation. You're doing something very non-standard, that goes aside of every other party flow: explain why!

I know this is for encrypted user share, but most wouldn't get it immediately. This needs to be explained, and why it is here, and what is the purpose of this check in the grand scheme of things in relation to that feature.

@ItayLevyOfficial